packages/material-ui/src/ImageListItem/ImageListItem.js

```diff
@@ -1,128 +1,94 @@
 import * as React from 'react';
+import { isFragment } from 'react-is';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import debounce from '../utils/debounce';
 import withStyles from '../styles/withStyles';
 import isMuiElement from '../utils/isMuiElement';
-import { ownerWindow } from '../utils';
+import ImageListContext from '../ImageList/ImageListContext';
 
 export const styles = {
   /* Styles applied to the root element. */
   root: {
-    boxSizing: 'border-box',
-    flexShrink: 0,
-  },
-  /* Styles applied to the `div` element that wraps the children. */
-  tile: {
     position: 'relative',
-    display: 'block', // In case it's not rendered with a div.
-    height: '100%',
-    overflow: 'hidden',
+    lineHeight: 0, // Fix masonry item spacing
+    // display: 'flex',
+    // flexDirection: 'column',
   },
-  /* Styles applied to an `img` element child, if needed to ensure it covers the tile. */
-  imgFullHeight: {
+  /* Styles applied to an `img` element to ensure it covers the item. */
+  img: {
+    objectFit: 'cover',
+    width: '100%',
     height: '100%',
```

Comment:
I believe this is the line that makes problems with the text below the image,not working with some images.
```suggestion
    maxHeight: '120px',
```

We got to have some max-height value which will make the images with bigger height to align with the smaller. Not sure about the specific value we should use there, but seems like using a value over 120px again breaks the scenario.. Should we maybe use `calc` for the maxWidth, based on whether there is a title bar inside? I've tested that with this changes, the text below image examples works in Chrome, Edge and Firefox.

Reply:
Also would suggest adding maxHeight only when there is a title bar, otherwise we don't need it, so we don't regress the other scenarios